ipld/src/cid_hashset.rs

```diff
@@ -0,0 +1,15 @@
+// Copyright 2019-2023 ChainSafe Systems
+// SPDX-License-Identifier: Apache-2.0, MIT
+
+use ahash::HashSet;
+use cid::Cid;
+
+#[derive(Default)]
+pub struct CidHashSet(HashSet<u64>);
+
+impl CidHashSet {
+    pub fn insert(&mut self, cid: &Cid) -> bool {
+        let hash = self.0.hasher().hash_one(cid.to_bytes());
```

Comment:
Pretty sure you hash a CID without calling `to_bytes()` (which copies the bytes).